models/credmark/protocols/lending/compound/compound_v2.py

```diff
@@ -59,51 +103,103 @@ def run(self, input: Token) -> dict:
 
         # converting the address to 'Address' type for safety
         comp_asset = contract.functions.markets(input.address.checksum).call()
-        tokencontract = Contract(
-            address=input.address.checksum, abi=ERC_20_TOKEN_CONTRACT_ABI)
-        symbol = self.try_or(lambda: tokencontract.functions.symbol().call())
-        decimals = tokencontract.functions.decimals().call()
+        token = Token(address=input.address.checksum)
+        symbol = token.symbol
+        decimals = token.decimals
         totalSupply = comp_asset[3]/pow(10, decimals)
         totalBorrows = comp_asset[6]/pow(10, decimals)
         output[symbol] = {'totalsupply': totalSupply, 'totalborrow': totalBorrows}
 
         return output
 
 
-@Model.describe(slug="compound-token-liability",
+class CompoundDebtInfo(DTO):

```

Comment:
This is a good start. I think we can potentially use this as a model for some additional data types like 

Pool(Contract)
LendingPool(Pool)
CompoundLendingPool(LendingPool)

Let's start thinking about this soon :)